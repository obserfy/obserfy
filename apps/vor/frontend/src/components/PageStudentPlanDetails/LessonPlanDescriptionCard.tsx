import { t, Trans } from "@lingui/macro"
import { FC, useState } from "react"
import { Button, Card, Flex } from "theme-ui"
import usePatchPlan from "../../hooks/api/plans/usePatchPlans"
import useVisibilityState from "../../hooks/useVisibilityState"
import { ReactComponent as CloseIcon } from "../../icons/close.svg"
import Icon from "../Icon/Icon"
import MarkdownEditor from "../MarkdownEditor/MarkdownEditor"
import MultilineDataBox from "../MultilineDataBox/MultilineDataBox"
import { Typography } from "../Typography/Typography"

const LessonPlanDescriptionCard: FC<{
  planId: string
  description?: string
}> = ({ description, planId }) => {
  const editor = useVisibilityState()

  if (editor.visible) {
    return (
      <DescriptionEditor
        lessonPlanId={planId}
        onDismiss={editor.hide}
        onSave={editor.hide}
        initialValue={description}
      />
    )
  }

  return (
    <Card mb={3} pb={2} sx={{ borderRadius: [0, "default"] }}>
      <MultilineDataBox
        label={t`Description`}
        value={description || ""}
        onEditClick={editor.show}
        placeholder="-"
      />
    </Card>
  )
}

const DescriptionEditor: FC<{
  initialValue?: string
  onDismiss: () => void
  onSave: () => void
  lessonPlanId: string
}> = ({ lessonPlanId, onDismiss, onSave, initialValue = "" }) => {
  const [value, setValue] = useState(initialValue)
  const { mutateAsync } = usePatchPlan(lessonPlanId)

  const handleSave = async () => {
    try {
      const result = await mutateAsync({ description: value })
      if (result?.ok) onSave()
      onSave()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <Card mb={3} sx={{ borderRadius: [0, "default"], overflow: "hidden" }}>
      <Flex py={3} sx={{ alignItems: "center" }}>
        <Typography.Body px={3} sx={{ fontWeight: "bold" }}>
          <Trans>Description</Trans>
        </Typography.Body>

        <Button variant="outline" ml="auto" p={1} mr={2} onClick={onDismiss}>
          <Icon as={CloseIcon} fill="danger" />
        </Button>

        <Button
          mr={3}
          sx={{ fontWeight: "bold", fontSize: 0 }}
          onClick={handleSave}
        >
          <Trans>Save</Trans>
        </Button>
      </Flex>

      <MarkdownEditor value={value} onChange={setValue} />
    </Card>
  )
}

export default LessonPlanDescriptionCard
