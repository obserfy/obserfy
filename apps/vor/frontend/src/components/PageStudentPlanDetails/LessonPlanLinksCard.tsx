import { Link, Button, Card, Flex } from "theme-ui"
import { Trans } from "@lingui/macro"
import { FC, useState } from "react"
import { Typography } from "../Typography/Typography"
import { GetPlanResponseBody } from "../../hooks/api/plans/useGetPlan"
import useDeleteLessonPlanLink from "../../hooks/api/plans/useDeleteLessonPlanLink"
import Icon from "../Icon/Icon"
import { ReactComponent as LinkIcon } from "../../icons/link.svg"
import LoadingIndicator from "../LoadingIndicator/LoadingIndicator"
import { ReactComponent as TrashIcon } from "../../icons/trash.svg"
import usePostNewLessonPlanLink from "../../hooks/api/plans/usePostNewLessonPlanLink"
import LinkInput from "../LinkInput/LinkInput"

const LessonPlanLinksCard: FC<{
  planId: string
  links: Array<{
    id: string
    url: string
    title?: string
    description?: string
    image?: string
  }>
}> = ({ links, planId }) => (
  <Card sx={{ borderRadius: [0, "default"] }} mb={3} pb={3}>
    <Typography.Body
      pt={3}
      mx={3}
      mb={1}
      color="textMediumEmphasis"
      sx={{ lineHeight: 1, fontSize: 1 }}
    >
      <Trans>Links</Trans>
    </Typography.Body>
    {links && links.length === 0 && (
      <Typography.Body m={3}>
        <Trans>No links attached yet</Trans>
      </Typography.Body>
    )}
    {links.map((link) => (
      <LinkItem key={link.id} link={link} lessonPlanId={planId} />
    ))}
    <UrlField lessonPlanId={planId} />
  </Card>
)

const LinkItem: FC<{
  link: GetPlanResponseBody["links"][0]
  lessonPlanId: string
}> = ({ link, lessonPlanId }) => {
  const [isLoading, setIsLoading] = useState(false)
  const deleteLink = useDeleteLessonPlanLink(link.id, lessonPlanId)

  const handleDelete = async () => {
    setIsLoading(true)
    try {
      await deleteLink.mutateAsync()
      setIsLoading(false)
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <Flex my={3} mr={3} sx={{ alignItems: "center", maxHeight: "100%" }}>
      <Link
        href={link.url}
        target="_blank"
        rel="noopener noreferrer"
        sx={{
          display: "flex",
          alignItems: "center",
          overflowX: ["auto", "hidden"],
        }}
      >
        <Icon as={LinkIcon} ml={3} />
        <Typography.Body
          mx={2}
          sx={{
            whiteSpace: "nowrap",
            display: "inline-block",
            textDecoration: "underline",
          }}
        >
          {link.url}
        </Typography.Body>
      </Link>
      <Button
        variant="outline"
        ml="auto"
        color="danger"
        px={2}
        backgroundColor="surface"
        sx={{ zIndex: 2, flexShrink: 0 }}
        onClick={handleDelete}
        disabled={isLoading}
      >
        {isLoading ? (
          <LoadingIndicator ml={1} />
        ) : (
          <Icon as={TrashIcon} fill="danger" />
        )}
      </Button>
    </Flex>
  )
}

const UrlField: FC<{ lessonPlanId: string }> = ({ lessonPlanId }) => {
  const [isLoading, setIsLoading] = useState(false)
  const postNewLink = usePostNewLessonPlanLink(lessonPlanId)
  const [url, setUrl] = useState("")

  async function sendPostNewLinkRequest() {
    setIsLoading(true)
    try {
      await postNewLink.mutateAsync({ url })
      setUrl("")
    } catch (e) {
      Sentry.captureException(e)
    } finally {
      setIsLoading(false)
    }
  }

  return (
    <LinkInput
      value={url}
      onChange={setUrl}
      isLoading={isLoading}
      onSave={sendPostNewLinkRequest}
      containerSx={{ mx: 3, mb: 3, mt: 2 }}
      inputSx={{ backgroundColor: "background" }}
    />
  )
}

export default LessonPlanLinksCard
