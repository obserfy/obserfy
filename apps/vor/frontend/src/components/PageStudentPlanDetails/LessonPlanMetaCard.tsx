import { FC, useState } from "react"
import { Box, Card, Flex } from "theme-ui"
import { useLingui } from "@lingui/react"
import { t } from "@lingui/macro"
import dayjs, { Dayjs } from "../../dayjs"
import usePatchPlan from "../../hooks/api/plans/usePatchPlans"
import DataBox from "../DataBox/DataBox"

import DatePickerDialog from "../DatePickerDialog/DatePickerDialog"
import Dialog from "../Dialog/Dialog"
import DialogHeader from "../DialogHeader/DialogHeader"
import Input from "../Input/Input"
import { useGetCurriculumAreas } from "../../hooks/api/useGetCurriculumAreas"
import Chip from "../Chip/Chip"
import useVisibilityState from "../../hooks/useVisibilityState"
import useInputState from "../../hooks/useInputState"

interface Props {
  planId: string
  date?: string
  areaId?: string
  title?: string
}
const LessonPlanMetaCard: FC<Props> = ({ planId, date, areaId, title }) => (
  <Card mb={3} sx={{ borderRadius: [0, "default"] }}>
    <DateDataBox value={date} lessonPlanId={planId} />
    <AreaDataBox value={areaId} lessonPlanId={planId} />
    <TitleDataBox value={title} lessonPlanId={planId} />
  </Card>
)

const DateDataBox: FC<{ value?: string; lessonPlanId: string }> = ({
  value,
  lessonPlanId,
}) => {
  const dialog = useVisibilityState()
  const patchPlan = usePatchPlan(lessonPlanId)

  const updateDate = async (date: Dayjs) => {
    try {
      await patchPlan.mutateAsync({ date })
      dialog.hide()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <>
      <DataBox
        label={t`Date`}
        onEditClick={dialog.show}
        value={value ? dayjs(value).format("D MMMM YYYY") : "N/A"}
      />
      {dialog.visible && (
        <DatePickerDialog
          isLoading={patchPlan.isLoading}
          defaultDate={dayjs(value)}
          onConfirm={updateDate}
          onDismiss={dialog.hide}
        />
      )}
    </>
  )
}

const AreaDataBox: FC<{ value?: string; lessonPlanId: string }> = ({
  value,
  lessonPlanId,
}) => {
  const classes = useGetCurriculumAreas()
  const [areaId, setAreaId] = useState(value)
  const patchPlan = usePatchPlan(lessonPlanId)
  const dialog = useVisibilityState()

  const updateArea = async () => {
    try {
      await patchPlan.mutateAsync({ areaId })
      dialog.hide()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <>
      <DataBox
        label={t`Related Area`}
        value={classes.data?.find(({ id }) => id === value)?.name || "Other"}
        onEditClick={dialog.show}
      />
      {dialog.visible && (
        <Dialog>
          <DialogHeader
            title={t`Change Area`}
            onAcceptText={t`Save`}
            onCancel={dialog.hide}
            onAccept={updateArea}
            loading={patchPlan.isLoading}
          />
          <Flex
            backgroundColor="background"
            sx={{ flexWrap: "wrap" }}
            p={3}
            pb={2}
          >
            {classes.data?.map(({ id, name }) => (
              <Chip
                mr={2}
                mb={2}
                key={id}
                text={name}
                onClick={() => setAreaId(id)}
                isActive={id === areaId}
              />
            ))}
          </Flex>
        </Dialog>
      )}
    </>
  )
}

const TitleDataBox: FC<{ value?: string; lessonPlanId: string }> = ({
  lessonPlanId,
  value,
}) => {
  const dialog = useVisibilityState()
  const [title, setTitle] = useInputState(value)
  const patchPlan = usePatchPlan(lessonPlanId)
  const { i18n } = useLingui()

  const updateTitle = async () => {
    try {
      await patchPlan.mutateAsync({ title })
      dialog.hide()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <>
      <DataBox label="Title" value={value} onEditClick={dialog.show} />
      {dialog.visible && (
        <Dialog>
          <DialogHeader
            title={t`Edit Title`}
            onCancel={dialog.hide}
            onAccept={updateTitle}
            loading={patchPlan.isLoading}
          />
          <Box backgroundColor="background" p={3}>
            <Input
              label={i18n._(t`Title`)}
              sx={{ width: "100%" }}
              value={title}
              onChange={setTitle}
            />
          </Box>
        </Dialog>
      )}
    </>
  )
}

export default LessonPlanMetaCard
