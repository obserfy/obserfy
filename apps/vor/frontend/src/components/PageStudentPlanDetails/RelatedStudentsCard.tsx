import { Trans } from "@lingui/macro"
import { FC } from "react"
import { Button, Card, Flex, Image } from "theme-ui"
import useDeleteRelatedStudent from "../../hooks/api/plans/useDeleteRelatedStudent"
import usePostNewRelatedStudents from "../../hooks/api/plans/usePostNewRelatedStudents"
import useVisibilityState from "../../hooks/useVisibilityState"
import { ReactComponent as TrashIcon } from "../../icons/trash.svg"
import Icon from "../Icon/Icon"
import StudentPickerDialog from "../StudentPickerDialog/StudentPickerDialog"
import StudentPicturePlaceholder from "../StudentPicturePlaceholder/StudentPicturePlaceholder"
import { Typography } from "../Typography/Typography"

interface Student {
  id: string
  name: string
  profileImageUrl?: string
}
const RelatedStudentsCard: FC<{
  planId: string
  studentId: string
  students: Student[]
}> = ({ planId, students, studentId }) => {
  const dialog = useVisibilityState()
  const postNewRelatedStudent = usePostNewRelatedStudents(planId)

  const handlePostNewStudent = async (value: Student[]) => {
    const payload = {
      studentIds: value.map(({ id }) => id),
    }
    try {
      await postNewRelatedStudent.mutateAsync(payload)
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  const otherStudents = students.filter(({ id }) => id !== studentId)

  return (
    <Card p={3} sx={{ borderRadius: [0, "default"] }}>
      <Flex>
        <Typography.Body color="textMediumEmphasis">
          <Trans>Other Related Students</Trans>
        </Typography.Body>
        <Button
          variant="outline"
          ml="auto"
          sx={{ color: "textMediumEmphasis", fontSize: 1 }}
          onClick={dialog.show}
        >
          <Trans>Add More</Trans>
        </Button>
      </Flex>
      {otherStudents.map(({ name, profileImageUrl, id }) => (
        <StudentListItem
          key={id}
          studentId={id}
          planId={planId}
          name={name}
          profilePictureUrl={profileImageUrl}
        />
      ))}
      {otherStudents.length === 0 && (
        <Typography.Body py={4} sx={{ textAlign: "center" }}>
          <Trans>No other students added yet.</Trans>
        </Typography.Body>
      )}

      {dialog.visible && (
        <StudentPickerDialog
          filteredIds={students.map(({ id }) => id)}
          onDismiss={dialog.hide}
          onAccept={handlePostNewStudent}
        />
      )}
    </Card>
  )
}

const StudentListItem: FC<{
  planId: string
  studentId: string
  name: string
  profilePictureUrl?: string
}> = ({ planId, studentId, profilePictureUrl, name }) => {
  const deleteRelatedStudent = useDeleteRelatedStudent(planId, studentId)

  const handleDeleteRelatedStudent = async () => {
    try {
      await deleteRelatedStudent.mutateAsync()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <Flex sx={{ alignItems: "center" }} mt={3}>
      {profilePictureUrl ? (
        <Image src={profilePictureUrl} sx={{ borderRadius: "circle" }} />
      ) : (
        <StudentPicturePlaceholder />
      )}
      <Typography.Body ml={3}>{name}</Typography.Body>
      <Button
        data-cy="delete-student"
        variant="text"
        ml="auto"
        onClick={handleDeleteRelatedStudent}
      >
        <Icon as={TrashIcon} />
      </Button>
    </Flex>
  )
}

export default RelatedStudentsCard
