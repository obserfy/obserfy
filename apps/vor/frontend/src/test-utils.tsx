/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
