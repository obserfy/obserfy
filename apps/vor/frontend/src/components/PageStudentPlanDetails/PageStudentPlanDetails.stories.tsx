import { FC } from "react"
import PageStudentPlanDetails from "./PageStudentPlanDetails"

export default {
  title: "Core/PageStudentPlanDetails",
  component: PageStudentPlanDetails,
  parameters: {
    componentSubtitle: "Just a simple PageStudentPlanDetails",
  },
}

export const Basic: FC = () => (
  <PageStudentPlanDetails planId="asfddsafdsadf" studentId="asdfasfd" />
)
