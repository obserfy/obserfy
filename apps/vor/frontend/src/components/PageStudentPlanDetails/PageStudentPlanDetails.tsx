import { FC } from "react"
import { Box, Button } from "theme-ui"
import { t, Trans } from "@lingui/macro"
import useGetPlan from "../../hooks/api/plans/useGetPlan"
import useDeletePlan from "../../hooks/api/plans/useDeletePlan"
import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"
import {
  STUDENT_OVERVIEW_URL,
  STUDENT_PLANS_URL,
  STUDENTS_URL,
} from "../../routes"
import dayjs from "../../dayjs"
import { ReactComponent as TrashIcon } from "../../icons/trash.svg"
import { useGetStudent } from "../../hooks/api/useGetStudent"
import TopBar from "../TopBar/TopBar"
import RelatedStudentsCard from "./RelatedStudentsCard"
import LessonPlanLinksCard from "./LessonPlanLinksCard"
import LessonPlanMetaCard from "./LessonPlanMetaCard"
import LessonPlanDescriptionCard from "./LessonPlanDescriptionCard"
import Icon from "../Icon/Icon"
import AlertDialog from "../AlertDialog/AlertDialog"
import { navigate } from "../Link/Link"
import useVisibilityState from "../../hooks/useVisibilityState"

interface Props {
  studentId: string
  planId: string
}
export const PageStudentPlanDetails: FC<Props> = ({ studentId, planId }) => {
  const { data: plan, isLoading: isLoadingPlan } = useGetPlan(planId)
  const { data: student } = useGetStudent(studentId)

  const navigateBack = () =>
    navigate(STUDENT_PLANS_URL(studentId, dayjs(plan?.date)))

  const breadcrumbs = [
    {
      text: t`Students`,
      to: STUDENTS_URL,
    },
    {
      text: student?.name?.split(" ")[0] ?? "",
      to: STUDENT_OVERVIEW_URL(studentId),
    },
    {
      text: t`Plans`,
      to: STUDENT_PLANS_URL(studentId, dayjs(plan?.date)),
    },
    { text: t`Details` },
  ]

  if (isLoadingPlan) {
    return (
      <Box sx={{ maxWidth: "maxWidth.sm" }} px={3} pb={3} mx="auto">
        <TopBar breadcrumbs={breadcrumbs} />
        <LoadingPlaceholder sx={{ width: "100%", height: 213 }} mb={3} />
        <LoadingPlaceholder sx={{ width: "100%", height: 129 }} mb={3} />
        <LoadingPlaceholder sx={{ width: "100%", height: 140 }} mb={3} />
      </Box>
    )
  }

  console.log(plan)

  return (
    <Box sx={{ maxWidth: "maxWidth.sm" }} pb={3} mx="auto">
      <TopBar breadcrumbs={breadcrumbs} />
      <Box px={[0, 3]}>
        <LessonPlanMetaCard
          planId={planId}
          date={plan?.date}
          areaId={plan?.areaId}
          title={plan?.title}
        />
        <LessonPlanDescriptionCard
          planId={planId}
          description={plan?.description}
        />
        <LessonPlanLinksCard planId={planId} links={plan?.links ?? []} />
        <RelatedStudentsCard
          studentId={studentId}
          planId={planId}
          students={plan?.relatedStudents ?? []}
        />
        <DeleteLessonPlanButton
          planId={planId}
          title={plan?.title}
          onDeleted={navigateBack}
        />
      </Box>
    </Box>
  )
}

const DeleteLessonPlanButton: FC<{
  planId: string
  title?: string
  onDeleted: () => void
}> = ({ planId, title, onDeleted }) => {
  const deletePlan = useDeletePlan(planId)
  const dialog = useVisibilityState()

  const handleDelete = async () => {
    try {
      await deletePlan.mutateAsync()
      onDeleted()
    } catch (e) {
      Sentry.captureException(e)
    }
  }

  return (
    <>
      <Button
        variant="outline"
        my={3}
        mr={[3, 0]}
        ml="auto"
        onClick={dialog.show}
        color="danger"
      >
        <Icon as={TrashIcon} mr={2} fill="danger" />
        <Trans>Delete</Trans>
      </Button>
      {dialog.visible && (
        <AlertDialog
          title={t`Delete plan?`}
          positiveText={t`Delete`}
          body={t`Are you sure you want to delete ${title}?`}
          onNegativeClick={dialog.hide}
          onPositiveClick={handleDelete}
        />
      )}
    </>
  )
}

export default PageStudentPlanDetails
