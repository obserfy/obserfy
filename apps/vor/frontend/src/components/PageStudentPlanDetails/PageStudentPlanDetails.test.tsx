import { advanceTo } from "jest-date-mock"
import { render } from "../../test-utils"
import { Basic } from "./PageStudentPlanDetails.stories"

describe("PageStudentPlanDetails", () => {
  it("should render correctly", () => {
    advanceTo(new Date(2018, 5, 27, 0, 0, 0)) // r

    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
