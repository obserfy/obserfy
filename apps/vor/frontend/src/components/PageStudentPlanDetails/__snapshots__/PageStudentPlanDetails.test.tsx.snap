// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageStudentPlanDetails should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 16px;
  padding-right: 16px;
  padding-bottom: 16px;
  margin-left: auto;
  margin-right: auto;
  max-width: 640px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  margin: 8px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: transparent;
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  border: none;
  padding: 4px;
}

.emotion-4:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow-x: auto;
}

.emotion-7 {
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-7:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-7:last-child>.splitter {
  display: none;
}

.emotion-8:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-9 {
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: inline-block;
  font-size: 13px;
  color: var(--theme-ui-colors-textMediumEmphasis);
  white-space: nowrap;
}

.emotion-16:last-child {
  color: var(--theme-ui-colors-textPrimary);
}

.emotion-16:last-child>.splitter {
  display: none;
}

.emotion-17 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 213px;
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 129px;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 140px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <a
          class="emotion-3"
          data-cy="back-button"
          href="/dashboard/students/plans?studentId=asdfasfd&date=2018-06-27T00:00:00.000Z"
        >
          <button
            class="emotion-4"
          >
            <div
              class="emotion-5"
            />
          </button>
        </a>
        <div
          class="emotion-6"
        >
          <a
            class="emotion-7"
            href="/dashboard/students"
          >
            <span
              class="emotion-8"
            >
              Students
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/students/overview?studentId=asdfasfd"
          >
            <span
              class="emotion-8"
            />
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <a
            class="emotion-7"
            href="/dashboard/students/plans?studentId=asdfasfd&date=2018-06-27T00:00:00.000Z"
          >
            <span
              class="emotion-8"
            >
              Plans
            </span>
            <span
              class="splitter emotion-9"
            >
              /
            </span>
          </a>
          <div
            class="emotion-16"
          >
            Details
            <span
              class="splitter"
            >
              /
            </span>
          </div>
        </div>
      </div>
      <div
        class="emotion-17"
      />
      <div
        class="emotion-18"
      />
      <div
        class="emotion-19"
      />
    </div>
  </div>
</div>
`;
